janusgraph-core/src/main/java/org/janusgraph/graphdb/query/index/IndexCandidate.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.query.index;
+
+import org.janusgraph.core.Cardinality;
+import org.janusgraph.core.attribute.Cmp;
+import org.janusgraph.graphdb.query.condition.Condition;
+import org.janusgraph.graphdb.query.condition.PredicateCondition;
+import org.janusgraph.graphdb.types.CompositeIndexType;
+import org.janusgraph.graphdb.types.IndexType;
+
+import java.util.Set;
+
+/**
+ * @author Boxuan Li (liboxuan@connect.hku.hk)
+ */
+public class IndexCandidate {
+    private static final double EQUAL_CONDITION_SCORE = 4;
+    private static final double OTHER_CONDITION_SCORE = 1;
+    private static final double CARDINALITY_SINGE_SCORE = 1000;
+    private static final double CARDINALITY_OTHER_SCORE = 1000;
+
+    private static final double ORDER_MATCH = 1;
+    private static final double ALREADY_MATCHED_ADJUSTOR = -1.1;
+
+    private final IndexType index;
+    private final Set<Condition> subCover;
+    private final Object subCondition;
+
+    // initialize with the worst possible score
+    private double score = Double.NEGATIVE_INFINITY;
+
+    public IndexCandidate(final IndexType index,
+                          final Set<Condition> subCover,
+                          final Object subCondition) {
+        this.index = index;
+        this.subCover = subCover;
+        this.subCondition = subCondition;
+    }
+
+    public IndexType getIndex() {
+        return index;
+    }
+
+    public Set<Condition> getSubCover() {
+        return subCover;
+    }
+
+    public Object getSubCondition() {
+        return subCondition;
+    }
+
+    public void calculateScoreBruteForce() {
+        score = 0.0;
+        for (final Condition c : subCover) {
+            score += getConditionBasicScore(c) + getIndexTypeScore(index);
+        }
+    }
```

Comment:
Move it to BruteForceIndexSelectionStrategy. You can just add necessary parameters and return a calculated score. I.e. `private double calculateScore(Set<Condition> subCover, IndexType index)`